lambda_function/aws/resources/cloudformation_handler.py

```diff
@@ -0,0 +1,118 @@
+                except Exception as e:
+                    logger.error(
+                        f"Failed list CloudFormation Stack, region: {region}, Stack statuses filter: {stack_status_filter}; {e}")
+                    self.skip_delete = True
+                    self.next_token = None
+                    break
+
+                self._handle_list_response(response, region)
+
+                self.next_token = response.get('NextToken')
+                if self.lambda_context.get_remaining_time_in_millis() < consts.REMAINING_TIME_TO_REINVOKE_THRESHOLD:
+                    # Lambda timeout is too close, should return checkpoint for next run
+                    return self._handle_close_to_timeout(region)
+
+            self._cleanup_regions(region)
+
+        return {'aws_entities': self.aws_entities, 'next_resource_config': None, 'skip_delete': self.skip_delete}
+
+    def _handle_list_response(self, list_response, region):
+        stacks = list_response.get('StackSummaries', [])
+        with ThreadPoolExecutor(max_workers=consts.MAX_UPSERT_WORKERS) as executor:
+            futures = [executor.submit(self.handle_single_resource_item,
+                                       region, stack.get("StackId")) for stack in stacks
+                                                                         if stack['StackStatus'] != 'DELETE_COMPLETE']
+            for completed_future in as_completed(futures):
+                result = completed_future.result()
+                self.aws_entities.update(result.get('aws_entities', set()))
+                self.skip_delete = result.get('skip_delete', False) if not self.skip_delete else self.skip_delete
+
+    def handle_single_resource_item(self, region, stack_id, action_type='upsert'):
+        entities = []
+        skip_delete = False
+        try:
+            stack_obj = {}
+            if action_type == 'upsert':
+                logger.info(f"Get CloudFormation Stack, id: {stack_id}")
+
+                aws_cloudformation_client = boto3.client("cloudformation", region_name=region)
+                stack_obj = aws_cloudformation_client.describe_stacks(StackName=stack_id).get("Stacks")[0]
+                stack_obj['StackResources'] = aws_cloudformation_client.describe_stack_resources(StackName=stack_id)\
+                    .get('StackResources')
+                stack_obj['Url'] = self._get_stack_console_url(stack_obj['StackId'])
+                template = aws_cloudformation_client.get_template(StackName=stack_id,
+                                                                  TemplateStage='Original')\
+                                                                  .get('TemplateBody')
+
+                # If template is a SAM proccessed JSON, converts to yaml
+                if isinstance(template,dict) or isinstance(template,OrderedDict):
+                    template = yaml.dump(json.loads(json.dumps(template)))
```

Comment:
Add some comment to explain all the loads / dumps purpose

Reply:
Try to make it more clear, let me know if it's better now